geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHash.java

```diff
@@ -65,50 +68,43 @@
   // have the advantage of saving us a lot of computation that would happen every time a new key was
   // added. if our internal implementation changes, these values may be incorrect. the tests will
   // catch this change. an increase in overhead should be carefully considered.
-  protected static final int BASE_REDIS_HASH_OVERHEAD = 232;
-  protected static final int HASH_MAP_VALUE_PAIR_OVERHEAD = 96;
-  protected static final int SIZE_OF_OVERHEAD_OF_FIRST_PAIR = 96;
+  protected static final int BASE_REDIS_HASH_OVERHEAD = 336;
+  protected static final int HASH_MAP_VALUE_PAIR_OVERHEAD = 47;
 
   private static final int defaultHscanSnapshotsExpireCheckFrequency =
       Integer.getInteger("redis.hscan-snapshot-cleanup-interval", 30000);
 
   private static final int defaultHscanSnapshotsMillisecondsToLive =
       Integer.getInteger("redis.hscan-snapshot-expiry", 30000);
 
-  private int HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS;
-  private int MINIMUM_MILLISECONDS_FOR_HSCAN_SNAPSHOTS_TO_LIVE;
+  private int HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS =
+      defaultHscanSnapshotsExpireCheckFrequency;
+  private int MINIMUM_MILLISECONDS_FOR_HSCAN_SNAPSHOTS_TO_LIVE =
+      defaultHscanSnapshotsMillisecondsToLive;
 
 
   @VisibleForTesting
-  public RedisHash(List<ByteArrayWrapper> fieldsToSet, int hscanSnapShotExpirationCheckFrequency,
+  public RedisHash(List<byte[]> fieldsToSet, int hscanSnapShotExpirationCheckFrequency,
       int minimumLifeForHscanSnaphot) {
-    this();
+    sizeInBytes = BASE_REDIS_HASH_OVERHEAD;
     HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS = hscanSnapShotExpirationCheckFrequency;
     MINIMUM_MILLISECONDS_FOR_HSCAN_SNAPSHOTS_TO_LIVE = minimumLifeForHscanSnaphot;
 
-    Iterator<ByteArrayWrapper> iterator = fieldsToSet.iterator();
+    hash = new Object2ObjectOpenCustomHashMap<>(fieldsToSet.size() / 2, ByteArrays.HASH_STRATEGY);
+    Iterator<byte[]> iterator = fieldsToSet.iterator();
     while (iterator.hasNext()) {
       hashPut(iterator.next(), iterator.next());
```

Comment:
I know it's outside the scope of the changes being made in this PR, but there's a possibility for a NoSuchElementException here if `fieldsToSet` doesn't have an even number of elements. Is it worth putting in a check for that, or something to handle it?

Reply:
That's such a good call out, Donal!  Could `fieldsToSet` ever be odd?  It seems like it could only happen if we intentionally create a new `RedisHash` from a test.  But maybe it's good to put a check anyway just in case?

Reply:
done.